packages/autorest.python/autorest/codegen/serializers/builder_serializer.py

```diff
@@ -557,6 +564,8 @@ def serialize_headers(self, builder: RequestBuilderType) -> List[str]:
                     self.code_model.is_legacy,
                 )
             )
+        if retval:
+            retval.insert(0, "# Construct headers")
```

Comment:
nit: i don't like array manipulation with inserts etc bc it's not readable. We can add a check of `if builder.parameters.headers` before going through the for loop, and if that's the case, add the comment

Reply:
make sense

Reply:
make sense